ocaml/xapi/xapi_pool.ml

```diff
@@ -315,10 +315,29 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
 
     match my_backend' with
     | Network_interface.Openvswitch ->
-      let my_controller = Db.Pool.get_vswitch_controller ~__context ~self:my_pool in
-      let controller = Client.Pool.get_vswitch_controller ~rpc ~session_id ~self:pool in
-      if my_controller <> controller && my_controller <> "" then
-        raise (Api_errors.Server_error(Api_errors.operation_not_allowed, ["vswitch controller address differs"]))
+      let my_legacy_controller = Db.Pool.get_vswitch_controller ~__context ~self:my_pool in
+      let remote_legacy_controller = Client.Pool.get_vswitch_controller ~rpc ~session_id ~self:pool in
+      if my_legacy_controller <> remote_legacy_controller && my_legacy_controller <> "" then
+        raise (Api_errors.Server_error(Api_errors.operation_not_allowed, ["vswitch controller address differs"]));
+
+      if List.length (Db.SDN_controller.get_all ~__context) > 0 then begin
```

Comment:
According to the exisiting logic of vswitch_controller, if I don't have a controller but remote has then it's OK, that's the intention.
But use match is much better :).
Thanks.